src/super_gradients/common/environment/monitoring/monitoring.py

```diff
@@ -0,0 +1,84 @@
+            StatAggregator(name="System/disk.io_write_mb", sampling_fn=disk.get_io_write_mb, aggregate_fn=average),
+            StatAggregator(name="System/memory.usage_percent", sampling_fn=virtual_memory.virtual_memory_used_percent, aggregate_fn=average),
+            StatAggregator(name="System/network.network_sent_mb", sampling_fn=network.get_network_sent_mb, aggregate_fn=average),
+            StatAggregator(name="System/network.network_recv_mb", sampling_fn=network.get_network_recv_mb, aggregate_fn=average),
+            StatAggregator(name="System/cpu.usage_percent", sampling_fn=cpu.get_cpu_percent, aggregate_fn=average),
+        ]
+
+        is_nvidia_lib_available = gpu.safe_init_nvidia_management_lib()
+        if is_nvidia_lib_available:
+            self.stat_aggregators += [
+                *GPUStatAggregatorIterator(
+                    name="System/gpu.memory_usage_percent", device_sampling_fn=gpu.get_device_memory_usage_percent, device_aggregate_fn=average
+                ),
+                *GPUStatAggregatorIterator(
+                    name="System/gpu.memory_allocated_percent", device_sampling_fn=gpu.get_device_memory_allocated_percent, device_aggregate_fn=average
+                ),
+                *GPUStatAggregatorIterator(name="System/gpu.usage_percent", device_sampling_fn=gpu.get_device_usage_percent, device_aggregate_fn=average),
+            ]
+
+            if extra_gpu_stats:
+                self.stat_aggregators += [
+                    *GPUStatAggregatorIterator(name="System/gpu.temperature_c", device_sampling_fn=gpu.get_device_temperature_c, device_aggregate_fn=average),
+                    *GPUStatAggregatorIterator(name="System/gpu.power_usage_w", device_sampling_fn=gpu.get_device_power_usage_w, device_aggregate_fn=average),
+                    *GPUStatAggregatorIterator(
+                        name="System/gpu.power_usage_percent", device_sampling_fn=gpu.get_device_power_usage_percent, device_aggregate_fn=average
+                    ),
+                ]
+
+        thread = threading.Thread(target=self._run, daemon=True, name="SystemMonitor")
+        thread.start()
+
+    def _run(self):
+        """Sample, aggregate and write the statistics regularly."""
+        while True:
+            for _ in range(self.n_samples_per_aggregate):
+                self._sample()
+                time.sleep(self.sample_interval)
+            self._aggregate_and_write()
+
+    def _sample(self):
+        """Sample the stat_aggregators, i.e. get the current value of each of them."""
+        for stat_aggregator in self.stat_aggregators:
+            stat_aggregator.sample()
+
+    def _aggregate_and_write(self):
+        """Aggregate and write the results."""
+        for stat_aggregator in self.stat_aggregators:
+            self.tensorboard_writer.add_scalar(tag=stat_aggregator.name, scalar_value=stat_aggregator.aggregate(), global_step=self.write_count)
```

Comment:
call the stat_aggregator.reset() here to make it more understandable

Reply:
Right, changed in 4be886f65b51c41b1b20074c04d5d19b8a87ac1a